hub/src/server/driverModel.ts

```diff
@@ -20,10 +20,31 @@ export interface PerformDeleteArgs {
   storageTopLevel: string;
 }
 
+export interface PerformStatArgs {
+  path: string;
+  storageTopLevel: string;
+}
+
+export interface StatResult {
+  exists: boolean;
+  lastModifiedDate?: number;
+  contentLength?: number;
+  contentType?: string;
+}
+
+export interface PerformRenameArgs {
+  path: string;
+  storageTopLevel: string;
```

Comment:
Why does a rename allow for renaming files across different top-level paths?

Reply:
More specifically, does rename support the notion of moving a file between buckets or prefixes?  Asking because if so, even allowing this sounds very dangerous.

Reply:
Using only one `storageTopLevel` arg makes sense to me. Will update that.

To give some context -- all the drivers just concat the these two strings together, and the relationship between end user buckets and this `storageTopLevel` is pretty implicit. The http handler and auth layers are currently responsible for enforcing anything like this, e.g. preventing cross-bucket file moves.